File: 3768-check-if-digits-are-equal-in-string-after-operations-i/3768-check-if-digits-are-equal-in-string-after-operations-i.py
class Solution:
    def hasSameDigits (self, s : str) -> bool:
        a = list (s)
        for i in range (len (a) - 1, 1, -1):
            for j in range (i):
                a [j] = str ((int (a [j]) + int (a [j + 1])) % 10)
        return a [0] == a [1]

File: 3768-check-if-digits-are-equal-in-string-after-operations-i/3768-check-if-digits-are-equal-in-string-after-operations-i.c
bool hasSameDigits (char* s) {
    for (int i = strlen (s) - 1; i >= 2; i--)
        for (int j = 0; j < i; j++)
            s [j] = (s [j] + s [j + 1]) % 10 + '0';
    return s [0] == s [1];
}